packages/nx/internal-testing-utils/assert-valid-migrations.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  MigrationsJson,
+  MigrationsJsonEntry,
+} from '../src/config/misc-interfaces';
+import * as path from 'path';
+
+export function assertValidMigrationPaths(json: MigrationsJson, root: string) {
+  Object.entries(json.generators).forEach(([generator, m]) => {
+    it(`should have valid path  generator: ${generator}`, () => {
+      validateMigration(m, root);
+    });
+  });
+
+  Object.entries(json.schematics ?? {}).forEach(([schematic, m]) => {
+    it(`should have valid path schematic: ${schematic}`, () => {
+      validateMigration(m, root);
+    });
+  });
+}
+
+function validateMigration(m: MigrationsJsonEntry, root: string) {
+  const impl = m.factory ?? m.implementation;
+  const implMember = impl.includes('#') ? impl.split('#')[1] : null;
+  const implPath = impl.includes('#') ? impl.split('#')[0] : impl;
```

Comment:
```suggestion
  const [implPath = impl, implMember = null] = impl.split('#')
```

idk if it's simpler, I just don't like checking impl.includes and impl.split twice 🤷